samples/cpp/tutorial_code/video/optical_flow/optical_flow_dense.cpp

```diff
@@ -0,0 +1,61 @@
+#include <iostream>
+#include <opencv2/core.hpp>
+#include <opencv2/highgui.hpp>
+#include <opencv2/imgproc.hpp>
+#include <opencv2/videoio.hpp>
+#include <opencv2/video.hpp>
+
+using namespace cv;
+using namespace std;
+
+int main()
+{
+    VideoCapture capture("vtest.avi");
+    if (!capture.isOpened()){
+        //error in opening the video input
+        cerr << "Unable to open file!" << endl;
+        return 0;
+    }
+
+    Mat frame1, prvs;
+    capture >> frame1;
+    cvtColor(frame1, prvs, COLOR_BGR2GRAY);
+
+    while(true){
+        Mat frame2, next;
+        capture >> frame2;
+        if (frame2.empty())
+            break;
+        cvtColor(frame2, next, COLOR_BGR2GRAY);
+
+        Mat flow(prvs.size(), CV_32FC2);
+        calcOpticalFlowFarneback(prvs, next, flow, 0.5, 3, 15, 3, 5, 1.2, 0);
+
+        // visualization
+        Mat flow_parts[2];
+        split(flow, flow_parts);
+        Mat magnitude, angle, magn_norm;
+        cartToPolar(flow_parts[0], flow_parts[1], magnitude, angle, true);
+        normalize(magnitude, magn_norm, 0.0f, 1.0f, NORM_MINMAX);
+        angle = angle / 360.0 * 255.0;
```

Comment:
@mehlukas 
Is this correct? I see an artificial red/blue boundary that doesn't match the flow vectors. Remember that OpenCV expects HSV hue as 0-180. The following has no red/blue boundary issue:

```
void flowViz(const cv::Mat& flow, cv::Mat& dst)
{
  cv::Mat uv[2];
  cv::split(flow, uv);
  cv::Mat magnitude, angle;
  cv::cartToPolar(uv[0], uv[1], magnitude, angle, true); 
  cv::normalize(magnitude, magnitude, 0.0, 1.0, cv::NORM_MINMAX);
  angle *= ((1.f / 360.f) * (180.f / 255.f));

  cv::Mat hsv_[3], hsv, hsv8;
  hsv_[0] = angle;
  hsv_[1] = cv::Mat::ones(angle.size(), CV_32F);
  hsv_[2] = magnitude;
  cv::merge(hsv_, 3, hsv);
  hsv.convertTo(hsv8, CV_8U, 255.0); 
  cv::cvtColor(hsv8, dst, cv::COLOR_HSV2BGR);
}
```

Reply:
Yes, you're right! Thanks for noticing.